sycl/source/detail/graph_impl.cpp

```diff
@@ -1601,16 +1610,14 @@ void modifiable_command_graph::begin_recording(
                           "differs from the graph device.");
```

Comment:
Unrelated to this PR specifically, but it seems we have duplicate error checks here?

Reply:
Removed the duplicated checks :)